pkg/deploy/api/types.go

```diff
@@ -121,6 +121,9 @@ type ExecNewPodHook struct {
 	// Volumes is a list of named volumes from the pod template which should be
 	// copied to the hook pod.
 	Volumes []string
+	// MaxRestarts specifies a number of allowed lifecycle pod container restarts
+	// before we treat the lifecycle hook as failed.
+	MaxRestarts int
```

Comment:
I wouldn't expect this here.  It seems closely related to `LifecycleHook.FailurePolicy` and needs to be validated against the `FailurePolicy` value, so I would have expected it under `LifecycleHook`.  Is this a better spot for a reason I'm not seeing?